core/src/codechain_machine.rs

```diff
@@ -87,6 +102,68 @@ impl CodeChainMachine {
     pub fn populate_from_parent(&self, header: &mut Header, parent: &Header) {
         header.set_score(parent.score().clone());
     }
+
+    fn verify_transaction_timelock<C: BlockInfo + TransactionInfo>(
+        transaction: &Transaction,
+        header: &Header,
+        client: &C,
+    ) -> Result<(), Error> {
+        let inputs = match transaction {
+            Transaction::AssetTransfer {
+                inputs,
+                ..
+            } => inputs,
+            _ => return Ok(()),
+        };
+        for input in inputs {
+            if let Some(timelock) = &input.timelock {
+                match timelock {
+                    Timelock::Block(value) if value > &header.number() => {
+                        return Err(StateError::Transaction(TransactionError::Timelocked {
+                            timelock: timelock.clone(),
```

Comment:
How about deriving Copy for Timelock and copying it instead of cloning?

Reply:
I fixed it. And it's much better.